recipes/taocpp-taopq/all/conanfile.py

```diff
@@ -0,0 +1,74 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+
+
+class TaoCPPTaopqConan(ConanFile):
+    name = "taocpp-taopq"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/taocpp/taopq"
+    description = "C++ client library for PostgreSQL"
+    topics = ("cpp17", "postgresql", "libpq", "data-base", "sql")
+    settings = "os", "build_type", "compiler", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    requires = "libpq/11.5"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _has_support_for_cpp17(self):
+        supported_compilers = [("apple-clang", 10), ("clang", 6), ("gcc", 7), ("Visual Studio", 15.7)]
+        compiler, version = self.settings.compiler, Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        if not self._has_support_for_cpp17():
+            raise ConanInvalidConfiguration("Taocpp Taopq requires C++17 or higher support standard."
+                                            " {} {} is not supported."
+                                            .format(self.settings.compiler,
+                                                    self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("taopq-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["TAOPQ_BUILD_TESTS"] = False
+            self._cmake.definitions["TAOPQ_INSTALL_DOC_DIR"] = "licenses"
+            if self.settings.os == 'Windows' and self.settings.compiler == 'Visual Studio':
+                self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
```

Comment:
This is only available since cmake 3.4

Reply:
Done! I've update the CMakeLists.txt wrapper.